packages/wear-os/internal/entities/geolocation.ts

```diff
@@ -0,0 +1,14 @@
+import { WatchRecord, WatchSample } from './index';
+import { WatchSensor } from '../watch-sensor';
+
+export class Geolocation extends WatchRecord {
+  constructor(samples: GeolocationSample[], detectedAt: Date) {
```

Comment:
Just wondering, is it typical for the geolocation samples to come in batches? Considering it is a sensor that will change less frequently, is there a way for the developers to indicate that they want to get a geolocation sample as soon as possible? Btw, any chance to make this compatible with the geolocation API so it works with the geofencing plugin, for example?

Reply:
> Just wondering, is it typical for the geolocation samples to come in batches? Considering it is a sensor that will change less frequently, is there a way for the developers to indicate that they want to get a geolocation sample as soon as possible? 

The developer can indicate a `batchSize` of 1 to receive the samples as soon as possible. There is no problem to do that.

> Btw, any chance to make this compatible with the geolocation API so it works with the geofencing plugin, for example?

I thought about it. This Geolocation record contains the latitude, longitude, and altitude properties, the other ones are missing (speed, direction, etc...). If we really need them, I could add by changing the plugin and the wear library.

Then, the Geofencing plugin uses the Geolocation record of the Geolocation plugin, so I see two options:

1. Both records contain the same properties (except the missing ones, as I said before), so if those properties are not used, we could trust in the casting that the geofencing tasks do to a Geolocation record (from Geolocation plugin).
2. Make the wear-os plugin to depend on the geolocation plugin and emit its Geolocation record.

Reply:
> > Just wondering, is it typical for the geolocation samples to come in batches? Considering it is a sensor that will change less frequently, is there a way for the developers to indicate that they want to get a geolocation sample as soon as possible?
> 
> The developer can indicate a `batchSize` of 1 to receive the samples as soon as possible. There is no problem to do that.

Great! That sounds perfect :)
 
> > Btw, any chance to make this compatible with the geolocation API so it works with the geofencing plugin, for example?
> 
> I thought about it. This Geolocation record contains the latitude, longitude, and altitude properties, the other ones are missing (speed, direction, etc...). If we really need them, I could add by changing the plugin and the wear library.

Is that very complicated or it is just to add the properties and fill the missing fields? If it is something like that, I would say that the more, the merrier 😛 

> Then, the Geofencing plugin uses the Geolocation record of the Geolocation plugin, so I see two options:
> 
> 1. Both records contain the same properties (except the missing ones, as I said before), so if those properties are not used, we could trust in the casting that the geofencing tasks do to a Geolocation record (from Geolocation plugin).
> 2. Make the wear-os plugin to depend on the geolocation plugin and emit its Geolocation record.

I would go with option 1. This is TypeScript, so strict "implements" is not needed. How about generating something that meets the GeolocationLike interface? Perhaps I could see if it is possible to update then the Geofencing plugin to use the GeolocationLike interface instead. There should be no problem, since we are not using the Record fields or the functions included in the Geolocation class. I would need to check that...

Reply:
I've implemented these changes in the new commits. It seems to work like a charm!